includes/wizards/class-engagement-wizard.php

```diff
@@ -95,12 +89,15 @@ public function register_api_endpoints() {
 	 * @return WP_REST_Response with the info.
 	 */
 	public function api_get_connection_status_settings() {
-		$configuration_manager = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'jetpack' );
-		return rest_ensure_response( $configuration_manager->get_mailchimp_connection_status() );
+		$jetpack_configuration_manager = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'jetpack' );
+		$wc_configuration_manager      = Configuration_Managers::configuration_manager_class_for_plugin_slug( 'woocommerce' );
+		$response                      = $jetpack_configuration_manager->get_mailchimp_connection_status();
+		$response['wcConnected']       = $wc_configuration_manager->is_active();
```

Comment:
[`$response` will be a `WP_Error` if connection is not good](https://github.com/Automattic/newspack-plugin/blob/583fe2687032b51ab57ee3a1bf41bc740ac7a762/includes/configuration_managers/class-jetpack-configuration-manager.php#L55), so this will cause fatal errors if you try and use a `WP_Error` like an array.

```suggestion
		if ( ! is_wp_error( $response ) ) {
			$response['wcConnected']   = $wc_configuration_manager->is_active();
		}
```

Reply:
Good catch https://github.com/Automattic/newspack-plugin/pull/274/commits/5a88e2de4628711f7aaaa4b4f1cb2a2ff8a6b7f1